internal/cli/server_upgrade.go

```diff
@@ -0,0 +1,362 @@
+	if err != nil {
+		c.ui.Output(
+			"Error upgrading server on %s: %s", c.platform, clierrors.Humanize(err),
+			terminal.WithErrorStyle())
+
+		c.ui.Output(upgradeFailHelp)
+
+		return 1
+	}
+
+	contextConfig := result.Context
+	advertiseAddr := result.AdvertiseAddr
+	httpAddr := result.HTTPAddr
+
+	// We update the context config if the server addr has changed between upgrades
+	if originalCfg.Server.Address != contextConfig.Server.Address {
+		originalCfg.Server.Address = contextConfig.Server.Address
+
+		if err := c.contextStorage.Set(ctxName, originalCfg); err != nil {
+			c.ui.Output(
+				"Error setting the CLI context: %s\n\n%s",
+				clierrors.Humanize(err),
+				errInstallRunning,
+				terminal.WithErrorStyle(),
+			)
+			return 1
+		}
+
+		c.ui.Output("Server address has changed after upgrade. This client will "+
+			"update its context with the new address, however any other clients "+
+			"using this server must manually update their server address listed below "+
+			"or find the address from `waypoint context list` which lists context %q address",
+			ctxName,
+			terminal.WithWarningStyle())
+	}
+
+	// Connect
+	log.Info("connecting to the server so we can verify the server upgrade", "addr", originalCfg.Server.Address)
+	conn, err = serverclient.Connect(ctx,
+		serverclient.FromContextConfig(originalCfg),
+		serverclient.Timeout(5*time.Minute),
+	)
+	if err != nil {
+		c.ui.Output(
+			"Error connecting to server: %s\n\n%s",
+			clierrors.Humanize(err),
+			errInstallRunning,
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+	client = pb.NewWaypointClient(conn)
+
+	resp, err = client.GetVersionInfo(ctx, &empty.Empty{})
+	if err != nil {
+		c.ui.Output(
+			"Error retrieving server version info: %s", clierrors.Humanize(err),
+			terminal.WithErrorStyle())
+		return 1
+	}
+
+	c.ui.Output("\nServer upgrade for platform %q context %q complete!",
+		c.platform, ctxName, terminal.WithSuccessStyle())
+
+	c.ui.Output("Waypoint has finished upgrading the server to version %q\n",
+		resp.Info.Version, terminal.WithSuccessStyle())
+
+	c.ui.Output(addrSuccess, advertiseAddr.Addr, "https://"+httpAddr,
+		terminal.WithSuccessStyle())
+
+	return 0
+}
+
+func (c *ServerUpgradeCommand) Flags() *flag.Sets {
+	return c.flagSet(0, func(set *flag.Sets) {
+		f := set.NewSet("Command Options")
+		f.BoolVar(&flag.BoolVar{
+			Name:    "auto-approve",
+			Target:  &c.confirm,
+			Default: false,
+			Usage:   "Confirm server upgrade.",
+		})
+		f.StringVar(&flag.StringVar{
+			Name:    "context-name",
+			Target:  &c.contextName,
+			Default: "",
+			Usage:   "Waypoint server context to upgrade.",
+		})
+		f.StringVar(&flag.StringVar{
+			Name:    "platform",
+			Target:  &c.platform,
+			Default: "",
+			Usage:   "Platform to upgrade the Waypoint server from.",
+		})
+		f.StringVar(&flag.StringVar{
+			Name:    "snapshot-name",
+			Target:  &c.snapshotName,
+			Default: "",
+			Usage:   "Platform to upgrade the Waypoint server from.",
```

Comment:
I use this wording on my `snapshot-name` flag on Uninstall:
```suggestion
			Usage:   "Filename to write the snapshot to.",
```

Reply:
Oops! Good catch on this string